pysoils/SoilGrids_downloader.py

```diff
@@ -0,0 +1,290 @@
+'''
+SoilGrids data downloader.
+==========================
+
+Authors: Mattia Mancini, Mingyuan Chen
+Created: 24-October-2023
+--------------------------
+
+Script that downlaods and stores soil data retrieved from the SoilGrids dataset.
+
+The SoilGrids dataset can be found at https://www.isric.org/explore/soilgrids;
+it was downloaded using the soilgrids 0.1.3 library (https://pypi.org/project/soilgrids/)
+Docs at https://www.isric.org/explore/soilgrids/faq-soilgrids and
+https://maps.isric.org/
+
+Soil data from SoilGrids requires
+to be processed as follows:
+    - 1. Compute depth weighted averages of all values: the raw data is
+            reported by depth, but we want averages for 0-60cm.
+    - 2. Rescale values based on the conversion factors reported at
+            https://www.isric.org/explore/soilgrids/faq-soilgrids
+
+'''
+
+# import libraries
+import time
+from soilgrids import SoilGrids
+import xarray as xr
+from pyproj import Transformer
+import numpy as np
+import kwargs
+import math
+from owslib.wcs import WebCoverageService
+# import os
+# import multiprocessing as mp
+
+# create Constants class
+class Constants:
```

Comment:
I think it would be better to move these into the SoilApp class as class constant variables.